lib/self-hosted-shared-dependencies.js

```diff
@@ -383,7 +388,7 @@ export async function build({
     }
 
     if (npmRegistry) {
-      Object.assign(options, npmRegistry);
+      Object.assign(options, { registry: npmRegistry });
```

Comment:
This change should not be included in this pull request, as it was part of #14

```suggestion
      Object.assign(options, npmRegistry);
```